deepspeed/runtime/zero/stage3.py

```diff
@@ -1379,7 +1382,11 @@ def __avg_scatter_grads(self, params_to_reduce: List[Parameter]) -> List[Tensor]
         global_world_size = dist.get_world_size()
         num_nodes = global_world_size // local_world_size
         if self.all2all_process_group is not None and num_nodes > 1:
-            grad_partitions_for_rank = all_to_all_quant_reduce(full_grads_for_rank, self.all2all_process_group)
+            grad_partitions_for_rank = (
+                all_to_all_loco_quant_reduce(params_to_reduce, self.all2all_process_group, self.zeropp_loco_param)
```

Comment:
1. This is because we want to use `param.xxx` within the `all_to_all_loco_quant_reduce` function to store, access, and modify the error feedback buffer corresponding to each parameter. For example, in the function, we use `p.inter_ef_buf[0]`. Since `full_grads_for_rank` is a list of tensors, assigning or maintaining attributes directly to it is not convenient.

2. Another reason is to minimize modifications to `stage3.py`. By restricting changes to our own functions, we reduce the risk of introducing unintended bugs.